Quotient/keyimport.cpp

```diff
@@ -0,0 +1,78 @@
+// SPDX-FileCopyrightText: 2024 Tobias Fella <tobias.fella@kde.org>
+// SPDX-License-Identifier: LGPL-2.0-or-later
+
+#include "keyimport.h"
+
+#include <QtEndian>
+#include <QDebug>
+
+#include "e2ee/cryptoutils.h"
+#include "connection.h"
+#include "room.h"
+
+#include "logging_categories_p.h"
+
+using namespace Quotient;
+
+KeyImport::KeyImport(QObject* parent)
+    : QObject(parent)
+{
+}
+
+KeyImport::Error KeyImport::importKeys(QString data, const QString& passphrase, Connection* connection)
+{
+    data.remove(QLatin1String("-----BEGIN MEGOLM SESSION DATA-----"));
+    data.remove(QLatin1String("-----END MEGOLM SESSION DATA-----"));
+    data.remove(QLatin1Char('\n'));
+    auto decoded = QByteArray::fromBase64(data.toLatin1());
+    if (decoded[0] != 1) {
+        qCWarning(E2EE) << "Wrong version byte";
+        return InvalidData;
+    }
+
+    if (decoded.size() < 69) {
+        qCWarning(E2EE) << "Data not long enough";
+        return InvalidData;
+    }
+
+    const auto salt = decoded.mid(1, 16);
+    const auto iv = decoded.mid(17, 16);
+    const auto rounds = qFromBigEndian<uint32_t>(decoded.mid(33, 4).data());
+    const auto payload = decoded.mid(37, decoded.size() - 69);
+    const auto expectedMac = decoded.right(32);
```

Comment:
That's quite a few magic numbers. Can we add some names/logic to at least some? Constants like `SaltOffset = VersionLength`, `IvOffset = SaltOffset + AesBlockSize` etc. would be great.